wallet-new/src/Cardano/Wallet/API/V1/Migration/Types.hs

```diff
@@ -48,14 +48,23 @@ instance Migrate from to => Migrate [from] [to] where
     eitherMigrate = mapM eitherMigrate
 
 -- | Migration from list to NonEmpty, as suggested by @akegalj.
-instance Migrate from to => Migrate [from] (NonEmpty to) where
-  eitherMigrate a = NE.fromList <$> mapM eitherMigrate a
+instance (Migrate from to, Typeable from, Typeable to) => Migrate [from] (NonEmpty to) where
+    eitherMigrate [] = Left . MigrationFailed $ mconcat
+        [ "Failed migrating "
+        , show (typeRep (Proxy @[from]))
+        , " to a non empty list of "
+        , show (typeRep (Proxy @to))
+        , " because the list was empty."
+        ]
+    eitherMigrate (x:xs) = (:|) <$> eitherMigrate x <*> mapM eitherMigrate xs
```

Comment:
the old instance could blow up with a `Right _|_` if we all `eitherMigrate []` :frowning: